build_functions/CMakeLists.txt

```diff
@@ -193,3 +193,22 @@ function(compileTargetAsC11 theTarget)
     set_target_properties(${theTarget} PROPERTIES CXX_STANDARD 11)
   endif()
 endfunction()
+
+set(LIBS_TO_BE_REAL_CHECKED CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+
+function (add_library())
+    _add_library(${ARGV})
+    get_target_property(lib_type ${ARGV0} TYPE)
+    if( (NOT ${CMAKE_CURRENT_LIST_DIR} MATCHES "deps") AND # ignore dependency libs
```

Comment:
>(NOT ${CMAKE_CURRENT_LIST_DIR} MATCHES "deps") [](http://example.com/codeflow?start=8&length=46)

MATCHES works on regular expressions. If the intention was to exclude everything coming from "deps" then maybe it should have some sort of .* before/after "deps". Like ".*deps.*". Otherwise I am afraid this will exclude only CMakeLists.txt in the root of "deps" folder (if we even have one there).

Same for next line: why would it exclude AAA_ut_lib since AAA_ut_lib if not a match for _ut_lib. It would be a match for .*_ut_lib. 

... not an expert in regex

./runs for the hills.